feat: display skipped files

dekelund opened the pull request:
Display and store skipped files as a fact, relative to the project.
The list enables us to track why some dependencies was not identified.

This is helpful, if we miss some dependency when running locally, or
if it's not identified when running a scheduled monitor.

Includes following changes:

- List first ten skipped files in display.ts, and the reason why it's skipped.
- A flag to toggle the list in display (--print-skipped-unmanaged-files)
- Sends skipped files to the registry

Ticket: TUN-96

dekelund commented on lib/scan.ts:
```diff
@@ -32,6 +35,14 @@ export async function scan(options: Options): Promise<PluginResponse> {
       return s !== null;
     }) as SignatureResult[];
 
+    filteredSignatures.forEach((s) => {
```
This might need to get updated after we merge PR 44 or PR 45

dekelund commented on lib/find.ts:
```diff
@@ -6,8 +6,19 @@ import { debug } from './debug';
+interface SkippedFile {
+  path: string;
+  reason: string;
+}
```
add an empty line between interfaces

anthogez reviewed (changes requested):
Requires further discussion before of move ahead with this one.

dekelund commented on lib/display/display.ts:
```diff
@@ -27,6 +27,42 @@ export function displaySignatures(scanResults: ScanResult[]): string[] {
+  }
+
+  const result: string[] = [chalk.whiteBright('Skipped files:\n')];
+
```
and that it's 10 instead of 15, 50 or 100 is just a coincident, it's just a random number. If we want to keep a limit, then feel free to suggest another number.

Commit:
fix: skip files greater than 2GB

Due to limitations with nodes Buffer, we agreed
to skip files greater than 2GB.

This is implemented by following changes:

- Add file size validation to find.ts
- Print number of files skipped, when running in debug mode

Ticket: TUN-96

Commit:
feat: display skipped files

Display and store skipped files as a fact, relative to the project.
The list enables us to track why some dependencies was not identified.

This is helpful, if we miss some dependency when running locally, or
if it's not identified when running a scheduled monitor.

Includes following changes:

- List first ten skipped files in display.ts, and the reason why it's skipped.
- A flag to toggle the list in display (--print-skipped-unmanaged-files)
- Sends skipped files to the registry

Ticket: TUN-96
